lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -799,6 +810,13 @@ modifyLedgerBody f (Cardano.Tx body keyWits) = Cardano.Tx body' keyWits
                         auxData
                         validity
 
+emptyTx
+    :: RecentEra era
+    -> Core.Tx (ShelleyLedgerEra era)
+emptyTx RecentEraConway = Core.mkBasicTx Core.mkBasicTxBody
+emptyTx RecentEraBabbage = Core.mkBasicTx Core.mkBasicTxBody
+emptyTx RecentEraAlonzo = Core.mkBasicTx Core.mkBasicTxBody
```

Comment:
I did manage to "shorten" this somehow, but not really sure if it's much better:

```suggestion
emptyTx :: RecentEra era -> Core.Tx (ShelleyLedgerEra era)
emptyTx = \case
    RecentEraConway  -> e
    RecentEraBabbage -> e
    RecentEraAlonzo  -> e
  where
    e :: forall e. Core.EraTx e => Core.Tx e
    e = Core.mkBasicTx Core.mkBasicTxBody
```

Reply:
Yes, I think there are qualities to both approaches. Per https://github.com/input-output-hk/cardano-wallet/pull/3773#discussion_r1144713258 I'd like to try out leaning a bit more towards duplication. Changing the internal style should be easy anyway.